diff --git a/makefile b/makefile
index ac7a5a4..2467043 100644
--- a/makefile
+++ b/makefile
@@ -125,7 +125,7 @@ ifndef FO_EXE
 	FIND_ORB_EXE=find_orb
 endif
 
-all: $(FO_EXE) $(FIND_ORB_EXE) fo_serve.cgi eph2tle$(EXE)
+all: $(FO_EXE) $(FIND_ORB_EXE) fo_serve.cgi eph2tle$(EXE) libfindorb.a
 
 CXXFLAGS+=-c -Wall -pedantic -Wextra -Werror $(ADDED_CXXFLAGS) -I $(INSTALL_DIR)/include
 
@@ -311,6 +311,11 @@ else
 endif
 	$(CP) $(INSTALL_FILES) $(IDIR)
 	$(CP) -n hints.txt $(IDIR)
+	$(MKDIR) $(LIB_DIR)
+	$(CP) libfindorb.a $(LIB_DIR)
+	$(CP) sigma.h $(INSTALL_DIR)/include
+	$(CP) mpc_obs.h $(INSTALL_DIR)/include
+	$(CP) monte0.h $(INSTALL_DIR)/include
 
 uninstall:
 ifdef EXE
@@ -346,3 +351,6 @@ make_deb:
 
 .cpp.o:
 	$(CXX) $(CXXFLAGS) $<
+
+libfindorb.a: $(OBJS)
+	ar crsv libfindorb.a $(OBJS)
